src/components/SettlementButton.tsx

```diff
@@ -1,143 +1,106 @@
+import type {Report} from '@src/types/onyx';
+import {EmptyObject} from '@src/types/utils/EmptyObject';
 import ButtonWithDropdownMenu from './ButtonWithDropdownMenu';
 import * as Expensicons from './Icon/Expensicons';
 import KYCWall from './KYCWall';
-import withNavigation from './withNavigation';
 
-const propTypes = {
-    /** Callback to execute when this button is pressed. Receives a single payment type argument. */
-    onPress: PropTypes.func.isRequired,
+type AnchorAlignment = {
+    horizontal: 'left' | 'right' | 'center';
+    vertical: 'top' | 'center' | 'bottom';
+};
 
-    /** Call the onPress function on main button when Enter key is pressed */
-    pressOnEnter: PropTypes.bool,
+type TriggerKYCFlow = (event: Event, iouPaymentType: string) => void;
 
-    /** Settlement currency type */
-    currency: PropTypes.string,
+type SettlementButtonOnyxProps = {
+    /** The last payment method used per policy */
+    nvpLastPaymentMethod?: Record<string, string>;
+};
 
+type SettlementButtonProps = SettlementButtonOnyxProps & {
+    /** Callback to execute when this button is pressed. Receives a single payment type argument. */
+    onPress: (paymentType: string) => void;
+    /** The route to redirect if user does not have a payment method setup */
+    enablePaymentsRoute: string;
+    /** Call the onPress function on main button when Enter key is pressed */
+    pressOnEnter?: boolean;
+    /** Settlement currency type */
+    currency?: string;
     /** When the button is opened via an IOU, ID for the chatReport that the IOU is linked to */
-    chatReportID: PropTypes.string,
-
+    chatReportID?: string;
     /** The IOU/Expense report we are paying */
-    iouReport: iouReportPropTypes,
-
-    /** The route to redirect if user does not have a payment method setup */
-    enablePaymentsRoute: PropTypes.string.isRequired,
-
-    /** Should we show the approve button? */
-    shouldHidePaymentOptions: PropTypes.bool,
-
+    iouReport?: OnyxEntry<Report> | EmptyObject;
     /** Should we show the payment options? */
-    shouldShowApproveButton: PropTypes.bool,
-
-    /** The last payment method used per policy */
-    nvp_lastPaymentMethod: PropTypes.objectOf(PropTypes.string),
-
+    shouldHidePaymentOptions?: boolean;
+    /** Should we show the payment options? */
+    shouldShowApproveButton?: boolean;
     /** The policyID of the report we are paying */
-    policyID: PropTypes.string,
-
+    policyID?: string;
     /** Additional styles to add to the component */
-    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
-
+    style?: StyleProp<ViewStyle>;
     /** Total money amount in form <currency><amount> */
-    formattedAmount: PropTypes.string,
-
+    formattedAmount?: string;
     /** The size of button size */
-    buttonSize: PropTypes.oneOf(_.values(CONST.DROPDOWN_BUTTON_SIZE)),
-
+    buttonSize?: ButtonSizeValue; // Replace with the actual sizes
```

Comment:
is this a todo?

Reply:
I think he meant this comment: "// Replace with the actual sizes"